signature bridge contract implementation

dharjeezy opened the pull request:
closes #50 

### Task Description

- [x] Implemented Signature Bridge

- [x]  Make use of low level cross contract calls to handler contracts

- [x] Ink crates version bump across contracts and lib

- [x] Test for Admin Set Resource for Token wrapper handler in signature bridge

- [x] Test for Set fee proposal for Token wrapper handler in signature bridge

dharjeezy committed:
signature bridge contract implementation

dharjeezy committed:
execute proposal

drewstone commented on contracts/signature_bridge/Cargo.toml:
```diff
@@ -0,0 +1,51 @@
+name = "signature_bridge"
+path = "lib.rs"
+crate-type = [
+	# Used for normal contract Wasm blobs.
```
mixed tabs and spaces

dharjeezy replied:
This was actually code generated, when i created the contract at first.

drewstone commented on contracts/signature_bridge/lib.rs:
```diff
@@ -0,0 +1,197 @@
+    }
+
+    /// Get the `chain_id_type` from bytes array.
+    pub fn get_chain_id_type(chain_id_type: &[u8]) -> u64 {
```
Can we maybe add these functions to webb-rs and export them so we aren't re-using this so often?

dharjeezy replied:
Alright

drewstone reviewed (changes requested):
Looks like a good start, can you make sure to add tests too?

dharjeezy committed:
increase test time outs

dharjeezy commented on contracts/signature_bridge/lib.rs:
```diff
@@ -0,0 +1,197 @@
+    /// Example:
+    ///    chain_type - 0x0401, chain_id - 0x00000001 (big endian)
+    ///    Result - [00, 00, 04, 01, 00, 00, 00, 01]
+    pub fn compute_chain_id_type(chain_id: u64, chain_type: &[u8]) -> u64 {
```
Alright

dharjeezy committed:
change vec to u8 bytes

dharjeezy commented on contracts/signature_bridge/lib.rs:
```diff
@@ -0,0 +1,197 @@
+        chain_id: u64,
+        chain_type: [u8; 2],
+        counts: Mapping<Vec<u8>, [u8; 32]>,
+        resource_id_to_handler_address: Mapping<Vec<u8>, AccountId>,
```
> Looks like a good start, can you make sure to add tests too?

Yes, I will surely add tests. But for the signature bridge implementation to be complete, I need to have implemented the handlers specified here #54 #53 and #52

dharjeezy committed:
add webb-proposals lib

dharjeezy committed:
use webb proposals TypedChainId
kill substrate contract node after start

dharjeezy committed:
chain id bytes

dharjeezy committed:
Merge branch 'main' of https://github.com/webb-tools/protocol-ink into dami/signature-bridge-contract

 Conflicts:
	tests/governed_token_wrapper/tokenWrapper.test.ts

dharjeezy committed:
include padding

dharjeezy committed:
Ink typed chain id

dharjeezy committed:
rust up component

dharjeezy committed:
without locked flag

dharjeezy committed:
update to CI

dharjeezy committed:
remove console log

dharjeezy committed:
Merge branch 'main' of https://github.com/webb-tools/protocol-ink into dami/signature-bridge-contract

dharjeezy committed:
use protocol ink lib

dharjeezy committed:
Merge branch 'main' of https://github.com/webb-tools/protocol-ink into dami/signature-bridge-contract

dharjeezy committed:
ink version changes
openbrush version changes
version changes needed for signature bridge to get instantiated/deployed properly

dharjeezy committed:
recent changes

dharjeezy commented on contracts/signature_bridge/lib.rs:
```diff
@@ -0,0 +1,197 @@
+    pub struct SignatureBridge {
+        governor: Vec<u8>,
+        proposal_nonce: u32,
+        chain_id: u64,
```
Alright. I get you now.

dharjeezy committed:
logs

dharjeezy committed:
latest

dharjeezy committed:
add 27 to last sig element

dharjeezy committed:
admin_set_resource_with_signature test works

dharjeezy committed:
Merge branch 'main' of https://github.com/webb-tools/protocol-ink into dami/signature-bridge-contract

 Conflicts:
	contracts/token_wrapper_handler/lib.rs
	package.json
	protocol-ink-lib/Cargo.toml
	tests/util.ts
	yarn.lock

dharjeezy committed:
execute proposal with signature test

drewstone reviewed (approved):
LGTM!

drewstone commented:
Yep, just good to fix it up imo.

On Fri, Jul 8, 2022 at 3:04 PM dharjeezy ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In contracts/signature_bridge/Cargo.toml
> <https://github.com/webb-tools/protocol-ink/pull/51#discussion_r917073492>
> :
>
> > +scale = { package = "parity-scale-codec", version = "3", default-features = false, features = ["derive"] }
> +scale-info = { version = "2", default-features = false, features = ["derive"], optional = true }
> +
> +ark-ff = { version = "^0.3.0", default-features = false }
> +ark-std = { version = "^0.3.0", default-features = false }
> +ark-bn254 = { version = "^0.3.0", default-features = false, features = [ "curve" ] }
> +ark-crypto-primitives = { version = "^0.3.0", features = ["r1cs"], default-features = false }
> +
> +arkworks-setups = { version = "1.1.1",  features = ["r1cs"], default-features = false }
> +arkworks-native-gadgets = { version = "1.0.0", default-features = false }
> +
> +[lib]
> +name = "signature_bridge"
> +path = "lib.rs"
> +crate-type = [
> +	# Used for normal contract Wasm blobs.
>
> This was actually code generated, when i created the contract at first.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/webb-tools/protocol-ink/pull/51#discussion_r917073492>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADELLF2QD6HQDQGFFDWOJT3VTB3SXANCNFSM53ARFWRQ>
> .
> You are receiving this because your review was requested.Message ID:
> ***@***.***>
>
